autoload/go/template.vim

```diff
@@ -10,12 +11,19 @@ function! go#template#create() abort
   let l:package_name = go#tool#PackageName()
 
   " if we can't figure out any package name(no Go files or non Go package
-  " files) from the directory create the template
-  if l:package_name == -1
+  " files) from the directory create the template or use the cwd
+  " as the name
+  if l:package_name == -1 && l:go_template_no_file != 1
     let l:template_file = get(g:, 'go_template_file', "hello_world.go")
     let l:template_path = go#util#Join(l:root_dir, "templates", l:template_file)
     exe '0r ' . fnameescape(l:template_path)
     $delete _
+  elseif l:package_name == -1 && l:go_template_no_file == 1
+    " cwd is now the dir of the package
+    let l:path = fnamemodify(getcwd(), ':t')
```

Comment:
I see what you mean, but I think it would be confusing to a reader (of the docs or code) if the word directory isn't mentioned anywhere. 

This change, essentially, guesses that you most likely want to use the name of the directory that a new package is created in as the name of that package.

Using package instead of directory in the docs means that the docs would have to say something like, "use the package name for the name for new packages".

That's confusing because package means two different things in the same sentence.

But something like, "use the directory name of the package as the name for new packages," makes it clear what the two different things are and how they relate.

This would be the first time the directory is mentioned in the docs, but it would also be the first time that it was relevant.